assets/js/src/admin.js

```diff
@@ -26,3 +26,120 @@
 		} );
 	} );
 } )( window, jQuery );
+
+(function (window, $) {
+	'use strict';
+
+	var RSADisablePlugin = {
+
+		els: {
+			dialog: document.getElementById( 'rsa-disable-dialog' ),
+			placeholderA: document.getElementById( 'rsa-operator-a' ),
+			placeholderB: document.getElementById( 'rsa-operator-b' )
+		},
+
+		variables: {
+			operatorA: 0,
+			operatorB: 0,
+			expectedAnswer: 0,
+			disablingURL: null
+		},
+
+		openDialog: function ( event ) {
+			event.preventDefault();
+			$( this.els.dialog ).dialog( 'open' );
+		},
+
+		getRandomNum: function () {
+			return Math.floor( Math.random() * 10 );
+		},
+
+		isExpectedAnswer: function () {
+			var userResult = parseInt( document.getElementById( 'rsa-user-result' ).value, 10 );
+
+			if ( userResult === this.variables.expectedAnswer ) {
+				return true;
+			}
+
+			return false;
+		},
+		dialogSettings: function () {
+			var self = this;
+
+			$( this.els.dialog ).dialog({
+				title: '⚠️ ' + rsaAdmin.strings.warning,
+				dialogClass: 'wp-dialog',
+				autoOpen: false,
+				draggable: false,
+				width: 'auto',
+				modal: true,
+				resizable: false,
+				buttons: [
+					{
+						text: rsaAdmin.strings.confirm,
+						click: function () {
+							$( this ).dialog( 'close' );
+
+							if ( self.isExpectedAnswer() ) {
+								$.ajax({
+									method: 'post',
+									data: {
+										nonce: rsaAdmin.nonce,
+										user: rsaAdmin.user,
+										action: 'rsa_network_disable'
+									},
+									url: ajaxurl
+								}).always(function() {
+									window.location.href = self.variables.disablingURL;
+								});
```

Comment:
handle failure here so the user knows it didn’t work. could be a redirect that ads a page message

Reply:
This is a call to a log method. I don't think it's relevant to the user if we failed to log the action.